src/main/java/forestry/core/utils/SimpleByteBuf.java

```diff
@@ -0,0 +1,149 @@
+package forestry.core.utils;
+
+import java.util.Arrays;
+
+import forestry.api.genetics.AlleleManager;
+import forestry.api.genetics.IAllele;
+import forestry.api.genetics.IAlleleRegistry;
+import forestry.api.genetics.IChromosome;
+import forestry.api.genetics.IChromosomeType;
+import forestry.api.genetics.ISpeciesRoot;
+import forestry.core.genetics.Chromosome;
+
+public class SimpleByteBuf {
+
+	private byte[] data;
+	private int index;
+
+	public SimpleByteBuf(int initialCapacity) {
+		this.data = new byte[initialCapacity];
+	}
+
+	public SimpleByteBuf() {
+		this(20);
+	}
+
+	public SimpleByteBuf(byte[] data) {
+		this.data = data;
+	}
+
+	public byte[] toByteArray(){
+		return data;
+	}
+
+	public void writeAllele(IAllele allele){
+		IAlleleRegistry alleleRegistry = AlleleManager.alleleRegistry;
+		writeVarInt(alleleRegistry.getAlleleID(allele));
+	}
+
+	public IAllele readAllele(){
+		IAlleleRegistry alleleRegistry = AlleleManager.alleleRegistry;
+		int id = readVarInt();
+		return alleleRegistry.getAlleleById(id);
+	}
+
+	private int getValidChromosomes(IChromosome[] chromosomes, IChromosomeType[] chromosomeTypes){
+		int types = 0;
+		for (IChromosomeType type : chromosomeTypes) {
+			int index = type.ordinal();
+			if (index >= chromosomes.length) {
+				types <<= 1;
+				continue;
+			}
+			types = (types << 1) + 1;
+		}
+		return types;
+	}
+
+	public void writeChromosomes(IChromosome[] chromosomes, ISpeciesRoot speciesRoot){
+		IChromosomeType[] chromosomeTypes = speciesRoot.getKaryotype();
+		writeVarInt(getValidChromosomes(chromosomes, chromosomeTypes));
+		for (IChromosomeType type : chromosomeTypes) {
+			int index = type.ordinal();
+			if (index >= chromosomes.length) {
+				continue;
+			}
+			IChromosome chromosome = chromosomes[type.ordinal()];
+			writeAllele(chromosome.getPrimaryAllele());
+			writeAllele(chromosome.getSecondaryAllele());
+		}
+	}
+
+	public IChromosome[] readChromosomes(ISpeciesRoot speciesRoot){
+		IChromosomeType[] chromosomeTypes = speciesRoot.getKaryotype();
+		int types = readVarInt();
+		IChromosome[] chromosomes = new IChromosome[chromosomeTypes.length];
+
+		String primarySpeciesUid = null;
+		String secondarySpeciesUid = null;
+
+		for (IChromosomeType type : chromosomeTypes) {
+			int index = type.ordinal();
+			int bit = (1 << chromosomeTypes.length - index - 1);
```

Comment:
What is this doing? I do not understand it.

Reply:
I still do not understand the bit logic here.

Reply:
I write a integer(`types`) to the byte array that I use to check if the array contains the data of a specific chromsome type.
I create this int with `getValidChromosomes()`. This bit logic is used to check if the chromsome type that I want to read from the array is contained in the array. 
A small example:
I want to get the first chromosome (SPECIES) from a bee genome that I have writen to the byte array.
First I read `types`, it has the value 8191(1111111111111).
Now I wan't to know if this byte array contains this chromsome type. If it does not It is broken and has to be fixed.
I want the first chromsome so I have the index 0 and we have 13 bee chromosomes.
`1 << 13 - 0 - 1` = 4096 (1000000000000)
`1000000000000 & 1111111111111` = 1000000000000
`1000000000000 == 1000000000000` = true

Reply:
How could we have a genome that is missing a chromosome type?

Reply:
Only if we add or remove a chromosome type I think. Should I remove it ?

Reply:
Yeah I think this can be removed. 
We should avoid adding or removing chromosomes except on major breaking versions.